### DISCUSSION TITLE
Xiaomi Miio or Xiaomi Miot

### DISCUSSION THREAD

**Discussion by PskNorz** [PiotrMachowski/lovelace-xiaomi-vacuum-map-card#546] [Q&A]
Hey i was trying to edit today my vaccum card and as i searching in here i saw Xiaomi Miot integration and a different card yaml configuration. So i am asking what is the difference between those to?

**Comment by PiotrMachowski** [comment]
[Xiaomi Miio](https://www.home-assistant.io/integrations/xiaomi_miio/) - built in integration
[Xiaomi Miot](https://github.com/al-one/hass-xiaomi-miot) - custom integration

**Reply by PskNorz** [reply]
i am using Miio for a long time, its better to switch to Miot? I m saying that because i have errors that i cant solve in Miio.

**Reply by PiotrMachowski** [reply]
No, it's (almost) always better to use built-in integrations is possible

**Reply by PskNorz** [reply]
Alright, it is possible to post some errors that i have got in my logs from the vaccum to guide a lil bit?

**Reply by PiotrMachowski** [reply]
You can do it, but I think it might be more effective to ask on HA forum or create an issue in HA repo

**Reply by PskNorz** [reply]
Ok first question, i have xiaomi vaccum map card ( frontend ) and xiaomi map extractor ( integration ) should i have them both installed?

1st error ( Xiaomi Miio ) - Svetlana is the name of the vaccum, i got only one device and its the vaccum.

Logger: homeassistant.components.xiaomi_miio
Source: helpers/update_coordinator.py:182
Integration: Xiaomi Miio (documentation, issues)
First occurred: February 10, 2023 at 3:33:48 AM (26 occurrences)
Last logged: 6:29:51 PM

Timeout fetching Svetlana data

-------

2nd error.

This error originated from a custom integration.

Logger: custom_components.xiaomi_cloud_map_extractor.camera
Source: custom_components/xiaomi_cloud_map_extractor/camera.py:278
Integration: xiaomi_cloud_map_extractor (documentation, issues)
First occurred: February 10, 2023 at 5:14:22 PM (29 occurrences)
Last logged: 2:28:57 PM

Unable to retrieve map data

------

3rd error

Logger: homeassistant.helpers.entity
Source: helpers/entity.py:548
First occurred: 12:47:03 PM (1 occurrences)
Last logged: 12:47:03 PM

Update of camera.xiaomi_cloud_map_extractor is taking over 10 seconds


I also got the blueprint from map extractor to disable updating the map when the vaccum is docked it keep give me those errors.

**Reply by PiotrMachowski** [reply]
> Ok first question, i have xiaomi vaccum map card ( frontend ) and xiaomi map extractor ( integration ) should i have them both installed?

Yes

You can ignore these errors, they might be caused by temporal wifi connection issues, it's also possible that the vacuum is just lagging.

**Reply by PskNorz** [reply]
ok so maybe i am just gonna hide them. Thank you for your help.
